xmhf/src/xmhf-core/include/arch/x86/xmhf-baseplatform-arch-x86.h

```diff
@@ -174,6 +174,12 @@ struct _guestmtrrmsrs {
 #define IA32_VMX_MSRCOUNT                       18
 
 #ifndef __ASSEMBLY__
+
+typedef struct {
+    bool guest_nmi_enable,  //If true, then the mHV can inject NMI to the current guest.
```

Comment:
This variable is never written to. For TrustVisor, should we set this to false when running PAL?
This variable should be modified in XMHF (by calling functions like nmi_enable() and nmi_disable()). It is very easy to cause race conditions (e.g. NMIs never delivered to guest) when this variable is not modified correctly.